python/ray/tune/tests/test_trial_runner_2.py

```diff
@@ -322,7 +322,11 @@ def testPauseResumeCheckpointCount(self):
         tempdir = tempfile.mkdtemp()
         self.addCleanup(shutil.rmtree, tempdir)
 
-        trial = Trial("__fake", checkpoint_config=CheckpointConfig(num_to_keep=2))
+        trial = Trial(
+            "__fake",
+            local_dir=tempdir,
```

Comment:
I needed to set `local_dir` to be the same as `TrialRunner._local_checkpoint_dir` for these tests to work. This is because `TrialRunner.resume` will set the trial checkpoint paths to be relative to its `_local_checkpoint_dir` and assumes a directory structure of:

```
<experiment logdir> == _local_checkpoint_dir
    | experiment-checkpoint.json
    | <trial.logdir>
        | checkpoint_00000
```

Previously, the trial checkpoints and experiment checkpoints were being saved in two separate places. On resume, the trial checkpoint dirs would be pointing at the wrong location.
```
<experiment logdir> == _local_checkpoint_dir
    | experiment-checkpoint.json

<default trial logdir>
    | checkpoint_00000
```

Is this trial directory structure okay to assume? Wondering if it will be too easy for future tests to make this mistake.

Reply:
I think it's fine to hardcode it provided there's a comment explaining it.